src/libraries/System.Text.Json/ref/System.Text.Json.csproj

```diff
@@ -17,6 +17,12 @@
     <Reference Include="System.Threading.Tasks.Extensions" />
     <Reference Include="Microsoft.Bcl.AsyncInterfaces" />
   </ItemGroup>
+  <ItemGroup Condition="'$(TargetFramework)' == '$(NetCoreAppCurrent)'">
+    <Compile Include="System.Text.Json.netcoreappcurrent.cs" />
```

Comment:
NIT: I would change the name to just be `System.Text.Json.netcoreapp.cs` and the other one to be `System.Text.Json.non-netcoreapp.cs`